content/08-mongodb/09-mongodb-indexes.mdx

```diff
@@ -0,0 +1,260 @@
+
+```
+db.comments.createIndex( {"name":1} )
+```
+
+If we take the same query with the explain planfrom before:
+
+```
+db.comments.find( { "name" : "Ramsay Bolton" } ).explain("executionStats")
+```
+we get:
+
+```
+{
+  queryPlanner: {
+    plannerVersion: 1,
+    namespace: 'sample_mflix.comments',
+    indexFilterSet: false,
+    parsedQuery: { name: { '$eq': 'Ramsay Bolton' } },
+    winningPlan: {
+      stage: 'FETCH',
+      inputStage: {
+        stage: 'IXSCAN',
+        keyPattern: { name: 1 },
+        indexName: 'name_1',
+        isMultiKey: false,
+        multiKeyPaths: { name: [] },
+        isUnique: false,
+        isSparse: false,
+        isPartial: false,
+        indexVersion: 2,
+        direction: 'forward',
+        indexBounds: { name: [ '["Ramsay Bolton", "Ramsay Bolton"]' ] }
+      }
+    },
+    rejectedPlans: []
+  },
+  executionStats: {
+    executionSuccess: true,
+    nReturned: 273,
+    executionTimeMillis: 0,
+    totalKeysExamined: 273,
+    totalDocsExamined: 273,
+    executionStages: {
+      stage: 'FETCH',
+      nReturned: 273,
+      executionTimeMillisEstimate: 0,
+      works: 274,
+      advanced: 273,
+      needTime: 0,
+      needYield: 0,
+      saveState: 0,
+      restoreState: 0,
+      isEOF: 1,
+      docsExamined: 273,
+      alreadyHasObj: 0,
+      inputStage: {
+        stage: 'IXSCAN',
+        nReturned: 273,
+        executionTimeMillisEstimate: 0,
+        works: 274,
+        advanced: 273,
+        needTime: 0,
+        needYield: 0,
+        saveState: 0,
+        restoreState: 0,
+        isEOF: 1,
+        keyPattern: { name: 1 },
+        indexName: 'name_1',
+        isMultiKey: false,
+        multiKeyPaths: { name: [] },
+        isUnique: false,
+        isSparse: false,
+        isPartial: false,
+        indexVersion: 2,
+        direction: 'forward',
+        indexBounds: { name: [ '["Ramsay Bolton", "Ramsay Bolton"]' ] },
+        keysExamined: 273,
+        seeks: 1,
+        dupsTested: 0,
+        dupsDropped: 0
+      }
+    }
+  }
+}
+```
+Comparing the same fields as the un-indexed query we see that the `stage` from the `winningPlan` is now `IXSCAN` which indicates an index was used.
+
+Additionally, we see that `totalDocsExamined` is now just the 273 documents where `name` is `"Ramsay Bolton"` and not the entire 50,303 documents. This increased efficiency can be see especially now with `executionTimeMillis` totaling 0ms. 
+
+Analyzing the explain plan on your most important queries will show you the performance of your index or if one needs to be created to increase the efficiency of your application. 
+
+## How to drop an index
+While the explain plan may show the need for an index, it can also do the opposite. For example, if an index is no longer needed or not adding that much improved performance, it may be in your best interest to drop that index and reserve the space or preserve some write performance. 
```

Comment:
```suggestion
While the explain plan may show the need for an index, it can also do the opposite. For example, if an index is no longer needed or not adding that much improved performance, it may be in your best interest to drop that index to reserve the space and gain some write performance. 
```